src/ensembl/src/content/app/entity-viewer/gene-view/GeneView.tsx

```diff
@@ -153,30 +147,58 @@ const GeneViewWithData = (props: GeneViewWithDataProps) => {
         <GeneViewTabs />
       </div>
       <div className={styles.geneViewTabContent}>
-        {props.selectedGeneTabName === GeneViewTabName.TRANSCRIPTS &&
+        {selectedTabs.primaryTab === GeneViewTabName.TRANSCRIPTS &&
           basePairsRulerTicks && (
             <DefaultTranscriptslist
               gene={props.gene}
               rulerTicks={basePairsRulerTicks}
             />
           )}
 
-        {props.selectedGeneTabName === GeneViewTabName.GENE_FUNCTION && (
+        {selectedTabs.primaryTab === GeneViewTabName.GENE_FUNCTION && (
           <GeneFunction gene={props.gene} />
         )}
 
-        {props.selectedGeneTabName === GeneViewTabName.GENE_RELATIONSHIPS && (
+        {selectedTabs.primaryTab === GeneViewTabName.GENE_RELATIONSHIPS && (
           <GeneRelationships />
         )}
       </div>
     </div>
   );
 };
 
-const mapStateToProps = (state: RootState) => ({
-  // FIXME: this will have to be superseded with a proper way we get ids
-  geneId: getEntityViewerActiveEnsObject(state)?.stable_id || null,
-  selectedGeneTabName: getEntityViewerActiveGeneTab(state)
-});
+const useGeneViewRouting = () => {
+  const dispatch = useDispatch();
+  const params: { [key: string]: string } = useParams();
+  const { genomeId, entityId } = params;
+  const { objectId: geneId } = parseFocusIdFromUrl(entityId);
+  const { search } = useLocation();
+  // TODO: discuss – is using URLSearchParams better than using the querystring package?
+  const view = new URLSearchParams(search).get('view');
+  const viewInRedux = useSelector(getGeneViewName);
+  const previousGenomeId = usePrevious(genomeId); // genomeId during previous render
+  const selectedTabs = useSelector(getSelectedGeneViewTabs);
```

Comment:
I thought we didn't want to use these redux hooks 🤔

Reply:
We are flexible :-)